### PR TITLE
Quick editorial pass.

### PR DESCRIPTION
**Pull request by dmcardle**
Tried to clarify some text, PTAL :)

For the backend server, I think the `ClientHelloInner..ServerHelloECHConf` notation is a little confusing (pedantically, shouldn't it be `...`?). Can't we just concatenate the two messages and call it a day?

### CODE REVIEW

**Review by chris-wood — APPROVED** [review]
The simplified notation LGTM. @cjpatton, please have look!

File: draft-ietf-tls-esni.md [hunk 1]
```diff
@@ -944,13 +944,11 @@ It then computes a string
     accept_confirmation =
         Derive-Secret(Handshake Secret,
                       "ech accept confirmation",
-                      ClientHelloInner..ServerHelloECHConf)
+                      ClientHelloInner || ServerHelloECHConf)
```
Comment (cjpatton) [inline draft-ietf-tls-esni.md:947]: Actually, the `..` notation is meant to denote the sequence of messages between `ClientHelloInner` and `ServerHelloECHConf`. This sequence is empty here, but I think we should use the notation nonetheless, in order to be consistent with RFC8446.

File: draft-ietf-tls-esni.md [hunk 2]
```diff
@@ -944,13 +944,11 @@ It then computes a string
...
 ~~~
 
 where Derive-Secret and Handshake Secret are as specified in {{RFC8446}},
-Section 7.1 and ClientHelloInner..ServerHelloECHConf refers to the sequence of
-handshake messages beginning with the first ClientHello and ending with
-ServerHelloECHConf. Finally, the backend server constructs its ServerHello
+Section 7.1. Finally, the backend server constructs its ServerHello
```
Comment (cjpatton) [inline draft-ietf-tls-esni.md:951]: If we revert the above change, don't forget to revert this change as well.

**Review by cjpatton — CHANGES_REQUESTED** [review]
- I like `..` because it's consistent with RFC 8446 (See the key schedule section.)
- `make` fails because of lines that exceed 80 characters.

File: draft-ietf-tls-esni.md (hunk 1 above)
Comment (chris-wood) [reply draft-ietf-tls-esni.md:947]: I don’t think this is necessary. || is more clear in this context given that the sequence is always empty.

Comment (cjpatton) [reply draft-ietf-tls-esni.md:947]: What about incorporating the CH1 hash? Isn't that also something that's implied by the `..` notation?

File: draft-ietf-tls-esni.md [hunk 3]
```diff
@@ -460,8 +460,8 @@ otherwise.
 
 To offer ECH, the client first chooses a suitable ECH configuration. To
 determine if a given `ECHConfig` is suitable, it checks that it supports the KEM
-algorithm identified by `ECHConfig.kem_id`, at least one KDF/AEAD algorithm
-identified by `ECHConfig.cipher_suites`, and the version of ECH indicated by
+algorithm identified by `ECHConfigContents.kem_id`, at least one KDF/AEAD algorithm
```
Comment (cjpatton) [inline draft-ietf-tls-esni.md:463]: In this section, it might be clearer to refer to `ECHConfig.contents.kem_id`, since `ECHConfig` has been defined, but `ECHConfigContents` only implicitly refers to `ECHConfig.contents`.

File: draft-ietf-tls-esni.md (hunk 1 above)
Comment (cjpatton) [reply draft-ietf-tls-esni.md:947]: In other words, it denotes the whole transcript from ClientHelloInner to ServerHelloECHConf, including the messages on the HRR path.

Comment (cjpatton) [reply draft-ietf-tls-esni.md:947]: (@chris-wood and I spoke offline and we agreed that the `..` notation should stay as-is.)

Comment (dmcardle) [reply draft-ietf-tls-esni.md:947]: I find the range syntax a bit confusing here. If there were messages in between, it's not clear to me which variant of those messages would be selected, or whether some transformation would be applied to them.

I don't feel really strongly about this. Happy to revert it, but to be fully pedantic, I want to point out that RFC 8446 uses three dots between messages. I'll go ahead and use three dots here, unless two dots has a different meaning that I don't understand.

File: draft-ietf-tls-esni.md (hunk 2 above)
Comment (dmcardle) [reply draft-ietf-tls-esni.md:951]: Done!

File: draft-ietf-tls-esni.md (hunk 3 above)
Comment (dmcardle) [reply draft-ietf-tls-esni.md]: Agreed, made this change for the whole "Offering ECH" section.

File: draft-ietf-tls-esni.md (hunk 1 above)
Comment (cjpatton) [reply draft-ietf-tls-esni.md]: Oops, three dots is right! https://tools.ietf.org/html/rfc8446#section-7.1

I think my dyslexic brain is removing  a dot.

File: draft-ietf-tls-esni.md [hunk 4]
```diff
@@ -591,10 +591,10 @@ ClientHello extensions can be computed in this way.
 In contrast, clients do not know the longest SNI value in the client-facing
 server's anonymity set without server input. For the "server_name" extension
 with length D, clients SHOULD use the server's length hint L
-(ECHCOnfig.maximum_name_length) when computing the padding as follows:
+(ECHConfigContents.maximum_name_length) when computing the padding as follows:
```
Comment (cjpatton) [inline draft-ietf-tls-esni.md:594]: Is `ECHConfig.config.maximum_name_length` more appropriate here?

File: draft-ietf-tls-esni.md [hunk 5]
```diff
@@ -591,10 +591,10 @@ ClientHello extensions can be computed in this way.
...
 
 1. If L >= D, add L - D bytes of padding. This rounds to the server's
-   advertised hint, i.e., ECHConfig.maximum_name_length.
+   advertised hint, i.e., ECHConfigContents.maximum_name_length.
```
Comment (cjpatton) [inline draft-ietf-tls-esni.md:597]: `ECHConfig.contents.maximum_name_length`?

File: draft-ietf-tls-esni.md [hunk 6]
```diff
@@ -622,7 +622,7 @@ usual, authenticating the connection for the true server name.
 #### Rejected ECH
 
 If the server used ClientHelloOuter, the client proceeds with the handshake,
-authenticating for ECHConfig.public_name as described in {{auth-public-name}}.
+authenticating for ECHConfigContents.public_name as described in {{auth-public-name}}.
```
Comment (cjpatton) [inline draft-ietf-tls-esni.md:625]: `ECHConfig.contents.public_name`?

File: draft-ietf-tls-esni.md [hunk 7]
```diff
@@ -942,11 +942,11 @@ It then computes a string
     accept_confirmation =
         Derive-Secret(Handshake Secret,
                       "ech accept confirmation",
-                      ClientHelloInner..ServerHelloECHConf)
+                      ClientHelloInner...ServerHelloECHConf)
```
Comment (cjpatton) [inline draft-ietf-tls-esni.md:945]: Nice catch!

File: draft-ietf-tls-esni.md [hunk 8]
```diff
@@ -458,9 +458,9 @@ otherwise.
 
 To offer ECH, the client first chooses a suitable ECH configuration. To
 determine if a given `ECHConfig` is suitable, it checks that it supports the KEM
-algorithm identified by `ECHConfig.kem_id`, at least one KDF/AEAD algorithm
-identified by `ECHConfig.cipher_suites`, and the version of ECH indicated by
-`ECHConfig.version`. Once a suitable configuration is found, the client selects
+algorithm identified by `ECHConfig.contents.kem_id`, at least one KDF/AEAD algorithm
+identified by `ECHConfig.contents.cipher_suites`, and the version of ECH indicated by
+`ECHConfig.contents.version`. Once a suitable configuration is found, the client selects
```
Comment (cjpatton) [inline draft-ietf-tls-esni.md:463]: Do these lines break 80 characters?

**Review by cjpatton — CHANGES_REQUESTED** [review]
Looks good, modulo minor nits!

File: draft-ietf-tls-esni.md (hunk 8 above)
Comment (dmcardle) [reply draft-ietf-tls-esni.md:463]: Yep, good catch! Reflowing the paragraph in a new commit.

File: draft-ietf-tls-esni.md (hunk 6 above)
Comment (dmcardle) [reply draft-ietf-tls-esni.md]: Done

File: draft-ietf-tls-esni.md (hunk 5 above)
Comment (dmcardle) [reply draft-ietf-tls-esni.md]: Done

File: draft-ietf-tls-esni.md (hunk 4 above)
Comment (dmcardle) [reply draft-ietf-tls-esni.md]: Seems reasonable, done.

File: draft-ietf-tls-esni.md [hunk 9]
```diff
@@ -528,16 +529,16 @@ The client then generates the HPKE encryption context and computes the
...
     enc, context = SetupBaseS(pkR,
                               "tls ech" || 0x00 || ECHConfig)
     payload = context.Seal(ClientHelloOuterAAD,
                            EncodedClientHelloInner)
 ~~~
 
 Note that the HPKE functions Deserialize and SetupBaseS are those which match
-`ECHConfig.kem_id` and the AEAD/KDF used with `context` are those which match
-the client's chosen preference from `ECHConfig.cipher_suites`. The `info`
+`ECHConfig.contents.kem_id` and the AEAD/KDF used with `context` are those which match
```
Comment (cjpatton) [inline draft-ietf-tls-esni.md:540]: This line and the one below it are longer than 80 characters. (Sorry if this is annoying ... lines that are longer than 80 characters cause `make` to fail. In fact, you should be getting emails from CI telling you that the build is failing.)

**Review by cjpatton — CHANGES_REQUESTED** [review]

File: draft-ietf-tls-esni.md (hunk 9 above)
Comment (dmcardle) [reply draft-ietf-tls-esni.md]: Aha, I hadn't run make in a few commits, my bad. Fixed now.

**Review by cjpatton — APPROVED** [review]
Thanks!
